garak/_config.py

```diff
@@ -115,14 +116,23 @@ def _nested_dict():
 # placeholder
 # generator, probe, detector, buff = {}, {}, {}, {}
 
+def _key_exists(d: dict, key: str) -> bool:
+    # Check for the presence of a key in a nested dict.
+    if not isinstance(d, dict):
+        return False
```

Comment:
I don't think there are not any current places where this would occur, however should also process any `list` that may contain a `dict` objects?

Reply:
Alternatively one could go the other way and `raise ValueError("_key_exists() can only process dicts")`. I think it's probably beneficial to pick one way (dicts+lists of dicts) or the other (dicts or gtfo)